### ISSUE TITLE
Alternative SeekBar styles

### ISSUE THREAD

**Issue by OxygenCobalt** [OxygenCobalt/Auxio#130]
This is an issue about (maybe) adding alternative SeekBar styles to the main playback view. This would be controlled via a setting, like other views.

There would be the following styles that could be chosen:
- **Normal:** Like the current SeekBar
- **Wavy:** The odd Material You SeekBars that show a wave when playing. This is a lot like the rounded covers option, where it's for people who want the "true" Material You experience. This will be derived from the Android 13 builds.
![image](https://user-images.githubusercontent.com/65370175/167061592-104c1630-cbdc-4431-9851-1b145f701ccd.png)
- **Tall:** A taller SeekBar that shows the current audio info. For example, `FLAC 320khz`. This is most analogous to Music Player GO's old SeekBar. This is primarily intended to placate requests like #119 in a way that does not clutter the normal playback view.
![image](https://user-images.githubusercontent.com/65370175/167061840-1f1813a6-a092-4b0e-b2af-acdeb50f7c55.png)

**Comment by OxygenCobalt** [comment]
Note: This feature is now held up by either the resolution of [this material-components issue](https://github.com/material-components/material-components-android/issues/2690) or me reverse-engineering the wavy style when the Android 13 source is released.

**Comment by OxygenCobalt** [comment]
Apparently, Howl Music Player was able to implement the wavy SeekBar. I may or may not ~~steal~~ *borrow* their implementation and see if I can graft it onto the `Slider` view from `material-components`.

**Comment by OxygenCobalt** [comment]
Updated this issue to include Smooth Seeking too.

**Comment by OxygenCobalt** [comment]
Smooth seeking should arrive in the app in the next version. It does have a very minor visual issue that I'm waiting on https://github.com/material-components/material-components-android/issues/2940 to resolve.

**Comment by OxygenCobalt** [comment]
Next version should have a much more efficient implementation of smooth seeking.

**Comment by Donkey-Doug** [comment]
It would be nice to have the timestap while seeking showing over the album art.

**Comment by OxygenCobalt** [comment]
That's already in the app @Donkey-Doug. When you seek the left duration is highlighted and shows where you are seeking to.

**Comment by Donkey-Doug** [comment]
But it is shown next to the seekbar and not above album art in the middle of the screen.

**Comment by OxygenCobalt** [comment]
Eh. I prefer it in it's current position.

**Comment by OxygenCobalt** [comment]
I have come to the sudden realization that the "Wavy" SeekBar looks awfully like a certain kind of cell. Google seems to have also realized this, and has updated their wavy SeekBar to use a different thumb type. I'll likely do the same when I try to implement it.

![image](https://user-images.githubusercontent.com/65370175/194374391-b6cc8bc8-d2ce-4175-a640-7a3b93ba2f6e.png)

**Comment by OxygenCobalt** [comment]
I found the drawable implementation used by AOSP, [SquigglyProgress.kt](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/packages/SystemUI/src/com/android/systemui/media/controls/ui/SquigglyProgress.kt). If I vendor the slider component I could implement this soon.

**Comment by OxygenCobalt** [comment]
!!!!!!

https://github.com/galaxygoldfish/waveslider

Seems to be compose-specific though...

**Comment by mahozad** [comment]
Hi,

I'm doing a self-promotion here :)
Please ignore/close it if you are not interested.

Would you like to use the [wavy-slider](https://github.com/mahozad/wavy-slider) in your app?

[Demo](https://mahozad.ir/wavy-slider).

Although it's for Jetpack Compose, there are ways to use Composables in Views and vice versa.

**Comment by OxygenCobalt** [comment]
While I appreciate your promotion, I can't really use any Jetpack Compose components @mahozad, even through a View shim. Bringing in all of compose for a single view would probably cost several megabytes, a cost I'm not willing to take unless I made the entire app use Jetpack Compose (Something I'm also not willing to do since the effort required would be extreme)

**Comment by OxygenCobalt** [comment]
Sliders are officially available in MDC as of the newest alpha build. This is now unblocked.

**Comment by OxygenCobalt** [comment]
Wait, only available in the progress indicators of all things. Sigh. Blocked still.

**Comment by JimmysMaths** [comment]
@OxygenCobalt I sent some money hope this will come in the future. I love eye candy.

**Comment by OxygenCobalt** [comment]
Prioritized @JimmysMaths, this is quite an old issue so I'm not fully sure how it's going to pan out now. Will see.

**Comment by OxygenCobalt** [comment]
This is your only allowed prioritization as a $4/mo sponsor. Will remain prioritized until you stop sponsoring @JimmysMaths. Will be done after #927 and #984 since those are prioritized by a $8/mo sponsor.

**Comment by OxygenCobalt** [comment]
Hi @JimmysMaths, since you stopped sponsoring this is technically not prioritized, however since that also was during a period where I was highly busy I'll still keep this prioritized now for the next month or so.

**Comment by JimmysMaths** [comment]
@OxygenCobalt Don't worry about it you can take it off priority.

**Comment by OxygenCobalt** [comment]
A wavy seekbar is in the app now and will arrive in v4.1, but not any bitrate info. I don't like the "large block style" anymore really, so I see it more likely that I integrate bitrate or sample rate elsewhere. Where I'm still unsure. Closing and spinning off the bitrate aspect.
